bitcoinkit/src/main/kotlin/io/horizontalsystems/bitcoinkit/managers/ApiManagerBtcCom.kt

```diff
@@ -0,0 +1,43 @@
+package io.horizontalsystems.bitcoinkit.managers
+
+import io.horizontalsystems.bitcoinkit.models.PublicKey
+import io.reactivex.Flowable
+import io.reactivex.functions.Function3
+import java.util.*
+
+interface IApiManager {
+    fun getBlockHashes(pubKey: PublicKey): Flowable<List<BlockResponse>>
+}
+
+class ApiManagerBtcCom constructor(private val apiRequester: IApiRequester, private val addressSelector: IAddressSelector) : IApiManager {
+
+    override fun getBlockHashes(pubKey: PublicKey): Flowable<List<BlockResponse>> {
+        val addresses = addressSelector.getAddressVariants(pubKey)
+        val requests = ArrayList<Flowable<List<BlockResponse>>>()
+
+        for (i in 0..2) {
```

Comment:
Here we can get requests val by mapping addresses val. Then zip the whole requests val. To distinct values you can use Flowable.map